pkg/controller/servicebindingrequest/reconciler_test.go

```diff
@@ -399,3 +399,116 @@ func TestEmptyApplicationSelector(t *testing.T) {
+		// First sbr reconcile request
+		req := reconcile.Request{
+			NamespacedName: types.NamespacedName{
+				Namespace: reconcilerNs,
+				Name:      sbrName1,
+			},
+		}
+
+		// Reconcile first sbr
+		res, err := r.Reconcile(req)
+		require.NoError(t, err)
+		require.False(t, res.Requeue)
+
+		// Get sbr after reconciling
+		namespacedName := types.NamespacedName{Namespace: reconcilerNs, Name: sbrName1}
+		sbrOutput, err := r.getServiceBindingRequest(namespacedName)
+		require.NoError(t, err)
+
+		// expected sbr assertion
+		expectedStatus := v1alpha1.BoundApplication{
+			GroupVersionKind: v1.GroupVersionKind{
+				Group:   deploymentsGVR.Group,
+				Version: deploymentsGVR.Version,
+				Kind:    "Deployment",
+			},
+			LocalObjectReference: corev1.LocalObjectReference{
+				Name: applicationResourceRef,
+			},
+		}
+
+		require.Equal(t, BindingReady, sbrOutput.Status.Conditions[0].Type)
+		require.Equal(t, corev1.ConditionTrue, sbrOutput.Status.Conditions[0].Status)
+		require.Equal(t, sbrName1, sbrOutput.Status.Secret)
+		require.Len(t, sbrOutput.Status.Applications, 1)
+		require.True(t, reflect.DeepEqual(expectedStatus, sbrOutput.Status.Applications[0]))
+
+		// Reconciling second sbr
+		// Second sbr reconcile request
+		req = reconcile.Request{
+			NamespacedName: types.NamespacedName{
+				Namespace: reconcilerNs,
+				Name:      sbrName2,
+			},
+		}
+
+		// Reconcile second sbr
+		res, err = r.Reconcile(req)
+		require.NoError(t, err)
+		require.False(t, res.Requeue)
+
+		// Get sbr after reconciling
+		namespacedName = types.NamespacedName{Namespace: reconcilerNs, Name: sbrName2}
+		sbrOutput, err = r.getServiceBindingRequest(namespacedName)
+		require.NoError(t, err)
+
+		// expected sbr assertion
+		expectedStatus = v1alpha1.BoundApplication{
+			GroupVersionKind: v1.GroupVersionKind{
+				Group:   deploymentsGVR.Group,
+				Version: deploymentsGVR.Version,
+				Kind:    "Deployment",
+			},
+			LocalObjectReference: corev1.LocalObjectReference{
+				Name: applicationResourceRef,
+			},
+		}
+
+		require.Equal(t, BindingReady, sbrOutput.Status.Conditions[0].Type)
+		require.Equal(t, corev1.ConditionTrue, sbrOutput.Status.Conditions[0].Status)
+		require.Equal(t, sbrName2, sbrOutput.Status.Secret)
+		require.Len(t, sbrOutput.Status.Applications, 1)
+		require.True(t, reflect.DeepEqual(expectedStatus, sbrOutput.Status.Applications[0]))
+
+		// Get applicationResourceRef deployment
+		resourceClient := r.dynClient.Resource(deploymentsGVR).Namespace(namespacedName.Namespace)
+		u, err := resourceClient.Get(applicationResourceRef, metav1.GetOptions{})
+		require.NoError(t, err)
+		dep := &appsv1.Deployment{}
+		err = runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, dep)
+		require.NoError(t, err)
+
+		// Assert SBR name with secretRef object in containers env var
+		// Expected secretRef=binding-request1
+		require.Equal(t, sbrName1, dep.Spec.Template.Spec.Containers[0].EnvFrom[0].SecretRef.LocalObjectReference.Name)
```

Comment:
Shouldn't `dep.Spec.Template.Spec.Containers[0].EnvFrom[0].SecretRef.LocalObjectReference.Name`
contain the secret name which in this case is `db-credentials`  - https://github.com/redhat-developer/service-binding-operator/pull/514/files#diff-14a37dadc390fc3ec23459efc088de8eR418

Reply:
As per my understanding, it should be the intermediate secret's name, [it is stated in the last  para of doc](https://github.com/redhat-developer/service-binding-operator/blob/master/examples/nodejs_postgresql/README.md#express-an-intent-to-bind-the-db-and-the-application)

https://github.com/redhat-developer/service-binding-operator/pull/514/files#diff-14a37dadc390fc3ec23459efc088de8eR418 this is an actual secret.

Reply:
Resolving this conversation.